src/utils/postRequestProcessing.js

```diff
@@ -1,20 +1,17 @@
-// Calculate ZScore
 const calculateZScore = (responseData) => {
-  Object.keys(responseData).forEach((gene) => {
-    const { mean, stdev } = responseData[gene].rawExpression;
+  const dataWithZScore = Object.entries(responseData).reduce((acc, [gene, value]) => {
```

Comment:
Previously the function modifies its input. When I was fixing the test for marker heatmap, it fails because it says with `can not assign zScore to read only property`. So I checked and it turns out this function is modifying its input object. I don't think that's a good idea, so I refactored the function to return a copy of the response object, with the calculations included.

Reply:
Nice!